lib/tmp.js

```diff
@@ -121,7 +121,11 @@ function _generateTmpName(opts) {
 
   // mkstemps like template
   if (opts.template) {
-    return opts.template.replace(TEMPLATE_PATTERN, _randomChars(6));
+    var template = opts.template;
+    // make sure that we prepend the tmp path if none was given
+    if (path.basename(template) == template)
```

Comment:
This will be still a breaking change.

Reply:
@raszi Yes, this is somewhat breaking. And there may still exist security concerns for templates that resemble paths...